### PR TITLE
Add an Insights gallery: 17 alternative data views

### PR DESCRIPTION
**Pull request by kamilon**
## What

Adds an **Insights gallery** — a new "Insights (views)" entry in the overflow menu that opens a hub of **17 alternative visualizations** of the monitored data. The hub loads the data once (repos via `ActivityService`, trend history via `MetricStore`, teams via `TeamService`, people via `PeopleService`) into an immutable `InsightsData` bundle and hands it to each view, so switching between views is instant.

This lets the user glance at the same fleet through many lenses and keep whichever they like. **Additive only** — the sole change to existing code is the menu entry in `lib/home_menu.dart`.

## The 17 views (`lib/views/`)

| Lens | Views |
|---|---|
| At-a-glance | **Pulse** (counters + aggregate trend), **Fleet health** (0–100 gauge with factor breakdown) |
| Attention / triage | **Bubble** (age×count, sized by score), **Triage quadrant** (staleness×load, fixed thresholds), **CI health** board, **Freshness** (what went quiet) |
| Composition | **Treemap** (by score), **Open-PR donut**, **PR-age histogram**, **Provider split** (GitHub vs ADO) |
| Time | **Heatmap** (repo×day), **Trend lines**, **Stacked activity** |
| People | **Review load** (who can unblock whom), **Contributor cloud** |
| Power-user | **Sortable repo table**, **Team radar** (spider) |

Shared helpers (colors, scales, buckets, `ViewScaffold`, `openUrl`) live in `lib/views/views_common.dart`.

## Notes

- Time-based views (heatmap, trends, stacked) are sparse until trend history accumulates across days; each shows a friendly "builds up over time" state. Opening Insights now also captures a snapshot, like Radar/Teams/Digest.
- `activityScore` blends activity **and attention** (failing CI / stale PRs raise it) — this is disclosed on the score-encoding views. People views are framed as collaboration/unblocking per the app's no-surveillance ethos; the contributor cloud discloses the 5-authors-per-repo cap.

## Review

Ran the review gate (code-review + rubber-duck) **before** opening and fixed everything they found — see commit `5ccda70`:
- 3 confirmed crashes (age-histogram `clamp` on short surfaces, stacked-area dropping duplicate-leaf repos, review-load `_initials` RangeError)
- truthful encodings (treemap split clamp, provider zero-bars, quadrant midpoint→fixed thresholds, fleet-health CI factor = true pass rate)
- honesty disclosures + concurrent hub load.

## Tests

`flutter analyze --fatal-infos` clean · `dart format --set-exit-if-changed .` clean · **260 tests pass**. `test/views_smoke_test.dart` renders every view populated + empty, plus edge cases (short surface, duplicate leaf names, odd/unicode display names, single-provider, single repo).

### CODE REVIEW

File: lib/views/provider_split_view.dart [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                          ),
+                        ),
+                        Expanded(
+                          flex: ado,
+                          child: Container(
+                            color: adoColor.withValues(alpha: 0.85),
+                          ),
+                        ),
+                      ],
+                    ),
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/provider_split_view.dart:128]: Fixed in 9bee084 — the bar now emits only non-zero segments, so a single-provider fleet renders 100:0 with no flex-0 Expanded.

File: lib/views/insights_hub_page.dart [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    return RefreshIndicator(
+      onRefresh: _load,
+      child: GridView.builder(
+        padding: const EdgeInsets.all(12),
+        gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+          maxCrossAxisExtent: 220,
+          mainAxisSpacing: 12,
+          crossAxisSpacing: 12,
+          childAspectRatio: 1.25,
+        ),
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/insights_hub_page.dart:237]: Fixed in 9bee084 — added AlwaysScrollableScrollPhysics to the hub GridView so pull-to-refresh works when it doesn't overflow.

File: lib/views/views_common.dart [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    required this.people,
+    required this.loadedAt,
+  });
+
+  final List<RepoActivity> activities;
+  final Map<String, List<MetricSnapshot>> history;
+  final List<Team> teams;
+  final Map<String, TeamRollup> rollups;
+  final List<Person> people;
+  final DateTime loadedAt;
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/views_common.dart:39]: Fixed in 9bee084 — InsightsData now copies inputs into unmodifiable collections in the constructor (const dropped), so views can't mutate the shared snapshot.

File: lib/views/views_common.dart [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+};
+
+/// Opens a repo URL in the external browser; silently no-ops on failure.
+Future<void> openUrl(String url) async {
+  final uri = Uri.tryParse(url);
+  if (uri == null) return;
+  if (await canLaunchUrl(uri)) {
+    await launchUrl(uri, mode: LaunchMode.externalApplication);
+  }
+}
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/views_common.dart:111]: Fixed in 9bee084 — openUrl now wraps canLaunchUrl/launchUrl in try/catch and logs, so a platform exception can't crash the tap handler.

File: lib/views/health_gauge_view.dart [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  Widget build(BuildContext context) {
+    final repos = data.healthy;
+    final now = DateTime.now();
+
+    // Each factor is a 0..1 score (1 = healthy).
+    // CI factor: pass rate among repos with a definitive result (success vs
+    // failure); unknown/running are excluded so all-unknown doesn't read as 100%.
+    final success = repos.where((a) => a.ciStatus == 'success').length;
+    final failure = repos.where((a) => a.ciStatus == 'failure').length;
+    final ci = (success + failure) == 0 ? 1.0 : success / (success + failure);
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/health_gauge_view.dart]: Fixed in 8ead5f2 — when no repo has a definitive CI result the CI factor is now omitted from the composite entirely (not defaulted to 1.0), so the score no longer reads as 100% CI passing with no CI data.

File: lib/views/views_common.dart [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+    required Map<String, List<MetricSnapshot>> history,
+    required List<Team> teams,
+    required Map<String, TeamRollup> rollups,
+    required List<Person> people,
+    required this.loadedAt,
+  }) : activities = List.unmodifiable(activities),
+       history = Map.unmodifiable(history),
+       teams = List.unmodifiable(teams),
+       rollups = Map.unmodifiable(rollups),
+       people = List.unmodifiable(people);
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/views_common.dart:32]: Fixed in 8ead5f2 — InsightsData now deep-copies: nested per-repo history lists are List.unmodifiable and each TeamRollup is rebuilt with a Set.unmodifiable contributor set, so the immutability contract holds through the nesting.

File: lib/views/views_common.dart [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+
+String ciLabel(String status) => switch (status) {
+  'success' => 'passing',
+  'failure' => 'failing',
+  'running' => 'running',
+  _ => 'unknown',
+};
+
+/// Opens a repo URL in the external browser; silently no-ops on any failure
+/// (bad URL, unsupported scheme, or a platform channel exception).
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/views_common.dart]: Fixed in 8ead5f2 — reworded the doc to 'logs and no-ops on any failure' to match the debugPrint.

File: lib/views/heatmap_view.dart [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+          ),
+          for (final d in days)
+            Container(
+              width: cell,
+              height: cell,
+              margin: const EdgeInsets.all(1),
+              decoration: BoxDecoration(
+                color: row.byDay.containsKey(d)
+                    ? heatColor((row.byDay[d]! / maxScore).clamp(0.05, 1.0))
+                    : Theme.of(context).colorScheme.surfaceContainerHighest,
```
[the remark this answers is not in the document]
Comment (kamilon) [reply lib/views/heatmap_view.dart]: Fixed in the latest commit — a day with a snapshot but zero activity now renders as background (empty), not a faint colored cell. The 0.05 minimum clamp only applies to strictly-positive scores.
